File: SeleniumPomLab-master-Reto/src/test/java/com/demoblaze/pages/HomePage.java
package com.demoblaze.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.ui.ExpectedConditions;
import java.time.Duration;

public class HomePage {
    private WebDriver driver;
    private WebDriverWait wait;

    private By laptopsCategory = By.linkText("Laptops");
    private By welcomeMessage = By.id("nameofuser");

    public HomePage(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    }

    public void clickLaptopsCategory() {
        wait.until(ExpectedConditions.elementToBeClickable(laptopsCategory)).click();
    }

    public String getWelcomeMessage() {
        return wait.until(ExpectedConditions.visibilityOfElementLocated(welcomeMessage)).getText();
    }
}

File: CypressPomLab-master-Reto3/Cypress-TypeScript/cypress/pages/HomePage.ts
class HomePage {
  private welcomeMessage = '#nameofuser';

  clickLaptopsCategory(): void {
    cy.contains('Laptops').click();
  }

  getWelcomeMessage(): Cypress.Chainable<JQuery<HTMLElement>> {
    return cy.get(this.welcomeMessage);
  }
}

export default new HomePage();
